Cookbook/Proposals/Danger-Integration.md

```diff
@@ -0,0 +1,195 @@
+# Introduce Danger
+
+* Author: Olivier Halligon
+* Review Manager: TBD
+
+## Introduction
+
+[Danger](https://danger.systems/ruby/) is a tool that is able to comment on your Pull Requests based on some rules that you can write yourself.
+
+This proposal suggests that we adopt Danger to automatically check cases that we missed in the past during some PR reviews.
+
+This proposal also present an initial rule where Danger will be useful for us. Note that we will add more rules in the future via separate proposals or discussions (see [Alternatives Considered](#alternatives-considered)); this proposal focuses only on the first integration.
+
+
+## Motivation
+
+We have seen some common mistakes happen multiple times in our past PRs, like:
+
+ * `(null)` references in `pbxproj` files
```

Comment:
have you seen those "Recovered References"? Could we also cover it?

Reply:
Good point. I'd need a fixture project containing some to spike if it's easy to detect and implement, but I think it should be.

I'm hoping it's not a special magic group shown by Xcode, but is considered as a normal group similar to any other in the `pbxproj` file format; if so, it would be easy enough to detect during the same loop as this grepping for null references

Reply:
yes it is a normal group created by Xcode after a build. would make sense to be part of this loop 👍

Reply:
```suggestion
 * `(null)` and `"Recovered References"` in `pbxproj` files
```